components/camel-solr/src/main/java/org/apache/camel/component/solr/SolrConfiguration.java

```diff
@@ -0,0 +1,532 @@
+    }
+
+    /**
+     * Uses the provided solr client to connect to solr. When this parameter is not specified, camel applies the
+     * following rules to determine the SolrClient. A CloudSolrClient should point to a zookeeper endpoint. Other
+     * clients point to a Solr endpoint. 1) when zkHost or zkChroot (=zookeeper root) parameter is set, then the
+     * CloudSolrClient is used. 2) when multiple hosts are specified in the uri (separated with a comma), then the
+     * CloudSolrClient (uri scheme is 'solrCloud') or the LBHttpSolrClient (uri scheme is not 'solrCloud') is used. 3)
+     * when the solr operation is INSERT_STREAMING, then the ConcurrentUpdateSolrClient is used. 4) otherwise, the
+     * HttpSolrClient is used.
+     */
+    public void setSolrClient(SolrClient solrClient) {
+        this.solrClient = solrClient;
+    }
+
+    public HttpClient getHttpClient() {
+        return httpClient;
+    }
+
+    /**
+     * Sets the http client to be used by the solrClient
+     */
+    public void setHttpClient(HttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    public Boolean getUseConcurrentUpdateSolrClient() {
+        return useConcurrentUpdateSolrClient;
+    }
+
+    void setUseConcurrentUpdateSolrClient(boolean useConcurrentUpdateSolrClient) {
+        this.useConcurrentUpdateSolrClient = useConcurrentUpdateSolrClient;
+    }
+
+    public SolrEndpoint getSolrEndpoint() {
+        return solrEndpoint;
+    }
+
+    void setSolrEndpoint(SolrEndpoint solrEndpoint) {
+        this.solrEndpoint = solrEndpoint;
+    }
+
+    private String getFirstUrlFrom(String url) throws Exception {
+        return getUrlListFrom(url).get(0);
+    }
+
+    private List<String> getUrlListFrom(String url) throws Exception {
+        // add scheme when required
+        List<String> urlList = Arrays
+                .asList(url.split(","))
+                .stream()
+                .map(s -> solrScheme.getScheme().concat(s))
+                .collect(Collectors.toList());
+        // validate url syntax via parsing in URL instance
+        for (String s : urlList) {
+            try {
+                // solrCloud requires addition of HTTP scheme to be able to consider it as a valid URL scheme
+                new URL(
+                        SolrScheme.SOLRCLOUD.equals(solrScheme) ? SolrScheme.SOLR.getScheme().concat(s) : s);
+            } catch (MalformedURLException e) {
+                throw new IllegalArgumentException(
+                        String.format(
+                                "Url '%s' not valid for endpoint with uri=%s",
+                                s,
+                                solrScheme.getUri()));
+            }
+        }
+        return urlList;
+    }
+
+    /**
+     * signature defines parameters deciding whether or not to share the solrClient - sharing allowed: same signature -
+     * sharing not allowed: different signature
+     */
+    public String getSignature() {
```

Comment:
These seems utility methods. I think they should be in a separated class instead that the configuration class.